### PR TITLE
Minor cleanup

### PR DESCRIPTION
**Pull request by Lewox**
Added missing @Override annotation.
Refactored deprecated code.
Removed unused MatcherFilter#externalContainer variable.
Finalized some class members.
Simplified some boolean conditions.
Replaced StringBuffer (thread-safe) with StringBuilder.
Fixed minor typos.

### CODE REVIEW

**Review by mati-uy — APPROVED** [review]
LGTM!

File: src/main/java/spark/Service.java [hunk 1]
```diff
@@ -683,7 +682,7 @@ public synchronized int activeThreadCount() {
      */
     public synchronized <T extends Exception> void exception(Class<T> exceptionClass, ExceptionHandler<? super T> handler) {
         // wrap
-        ExceptionHandlerImpl wrapper = new ExceptionHandlerImpl<T>(exceptionClass) {
+        ExceptionHandlerImpl<T> wrapper = new ExceptionHandlerImpl<T>(exceptionClass) {
```
Comment (mauirace) [inline src/main/java/spark/Service.java:685]:
```suggestion
        ExceptionHandlerImpl<T> wrapper = new ExceptionHandlerImpl<>(exceptionClass) {
```

File: src/main/java/spark/route/Routes.java [hunk 2]
```diff
@@ -16,11 +16,7 @@
  */
 package spark.route;
 
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.HashMap;
-import java.util.List;
-import java.util.Map;
+import java.util.*;
```
Comment (mauirace) [inline src/main/java/spark/route/Routes.java:19]: 🤔  wildcard?

**Review by mauirace — APPROVED** [review]
Just some minor comments. LGTM!

File: src/main/java/spark/Service.java (hunk 1 above)
Comment (Lewox) [reply src/main/java/spark/Service.java:685]: Agreed, I'll include it in the next PR.

File: src/main/java/spark/route/Routes.java (hunk 2 above)
Comment (Lewox) [reply src/main/java/spark/route/Routes.java:19]: As per the author request, IntelliJ is used.

**Pull request merged**